### DISCUSSION TITLE
I cant seem to connect to the glsp server on windows 11 for the node-json-vscode configuration

### DISCUSSION THREAD

**Discussion by samadnasir23** [eclipse-glsp/glsp#1090] [Q&A]
GLSP-Server: [GLSP-Server]:Startup completed

extensionHostProcess.js:108
[JsonrpcGLSPClient] Connection to server is erroring. Shutting down server. (1) [Error: connect ECONNREFUSED 127.0.0.1:5007
	at TCPConnectWrap.afterConnect [as oncomplete] (node:ne…]
extensionHostProcess.js:108
[JsonrpcGLSPClient] Connection to server is erroring. Shutting down server. (1) [Error: connect ECONNREFUSED 127.0.0.1:5007
	at TCPConnectWrap.afterConnect [as oncomplete] (node:ne…]
extensionHostProcess.js:108
arg1:
(1) [Error: connect ECONNREFUSED 127.0.0.1:5007\n\tat TCPConnectWrap.afterConnect [as oncomplete] (node:ne…]
f @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:108:6
e @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:105482
<anonymous> @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:105350
error @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\@eclipse-glsp\protocol\src\jsonrpc\glsp-jsonrpc-client.ts:66:17
handleConnectionError @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\@eclipse-glsp\protocol\src\jsonrpc\base-jsonrpc-glsp-client.ts:134:27
<anonymous> @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\@eclipse-glsp\protocol\src\jsonrpc\base-jsonrpc-glsp-client.ts:128:41
invoke @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\vscode-jsonrpc\lib\events.js:62:39
fire @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\vscode-jsonrpc\lib\events.js:120:36
readErrorHandler @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\vscode-jsonrpc\lib\main.js:232:22
invoke @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\vscode-jsonrpc\lib\events.js:62:39
fire @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\vscode-jsonrpc\lib\events.js:120:36
fireError @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\vscode-jsonrpc\lib\messageReader.js:105:27
<anonymous> @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\vscode-jsonrpc\lib\messageReader.js:150:51
emit @ events:525:35
emitErrorNT @ internal/streams/destroy:157:8
emitErrorCloseNT @ internal/streams/destroy:122:3
processTicksAndRejections @ internal/process/task_queues:83:21
◀ TickObject ▶
init @ internal/inspector_async_hook:25:19
emitInitNative @ internal/async_hooks:202:43
emitInitScript @ internal/async_hooks:505:3
nextTick @ internal/process/task_queues:133:5
<anonymous> @ electron/js2c/utility_init:2:1366
onDestroy @ internal/streams/destroy:96:15
Socket._destroy @ net:763:5
_destroy @ internal/streams/destroy:102:25
destroy @ internal/streams/destroy:64:5
afterConnect @ net:1256:10
callbackTrampoline @ internal/async_hooks:130:17
◀ TCPCONNECTWRAP ▶
init @ internal/inspector_async_hook:25:19
emitInitNative @ internal/async_hooks:202:43
internalConnect @ net:991:26
defaultTriggerAsyncIdScope @ internal/async_hooks:464:18
emitLookup @ net:1156:9
onlookup @ dns:111:8
callbackTrampoline @ internal/async_hooks:130:17
◀ GETADDRINFOREQWRAP ▶
init @ internal/inspector_async_hook:25:19
emitInitNative @ internal/async_hooks:202:43
lookup @ dns:201:21
<anonymous> @ net:1133:5
defaultTriggerAsyncIdScope @ internal/async_hooks:464:18
lookupAndConnect @ net:1132:3
Socket.connect @ net:1070:5
start @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\@eclipse-glsp\vscode-integration\src\quickstart-components\socket-glsp-vscode-server.ts:92:21
activate @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\tasklist-vscode\extension\src\tasklist-extension.ts:67:19
processTicksAndRejections @ internal/process/task_queues:96:5
◀ await ▶
activate @ c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\tasklist-vscode\extension\src\index.ts:21:29
eb @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:13102
db @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:12816
<anonymous> @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:10913
◀ Promise.then ▶
bb @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:10832
$ @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:9989
◀ await ▶
actualActivateExtension @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:7100
n @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:6219
m @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:6180
l @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:5637
p @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:5564
k @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:5117
j @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:5072
<anonymous> @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:4039
h @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:4027
activateByEvent @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:3704
U @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:8350
<anonymous> @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:19776
processTicksAndRejections @ internal/process/task_queues:96:5
◀ Promise.then ▶
$activateByEvent @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:116:19762
N @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:114:7983
M @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:114:7749
H @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:114:6830
G @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:114:5906
<anonymous> @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:114:4736
w @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1902
fire @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2119
fire @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:79:13931
<anonymous> @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:130:30355
w @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1902
fire @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2119
fire @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:79:13931
<anonymous> @ c:\Users\WORK\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:130:28635
emit @ events:513:28
MessagePortMain._internalPort.emit @ electron/js2c/utility_init:2:367
callbackTrampoline @ internal/async_hooks:130:17
rejected promise not handled within 1 second: Error: Connection got disposed.
extensionHostProcess.js:108
stack trace: Error: Connection got disposed.
	at Object.dispose (c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at c:\Users\WORK\Desktop\Project\glsp-examples\project-templates\node-json-vscode\glsp-client\node_modules\@eclipse-glsp\protocol\lib\jsonrpc\base-jsonrpc-glsp-client.js:73:24
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
<img width="728" alt="Screenshot 2023-08-13 130647" src="https://github.com/eclipse-glsp/glsp/assets/123887191/c5dd6075-0e47-4e40-9f44-470ef54cdb26">

**Comment by planger** [comment]
Unfortunately, I don't have a Windows 11 machine for testing. In Windows 10, though, I can't reproduce this issue with the release and current master.

Which launch configuration did you open?
Which GLSP and VS Code version are you using?

Thank you!

**Reply by rhumbertgz** [reply]
@planger I get the same error but on MacOS.  Like @samadnasir23, I running the node-json-vscode template project. If I run the project on a linux container it works but then launching the extension takes a while because it has to open inside a container.

Node: 18.16.0
Yarn:  1.22.19
VSCode: 1.81.1
Laptop: Apple M1 Max
MacOS: 13.5.1 (22G90)

```
GLSP-Server: [GLSP-Server]:Startup completed
[JsonrpcGLSPClient] Connection to server is erroring. Shutting down server. (1) [Error: connect ECONNREFUSED 127.0.0.1:5007
	at TCPConnectWrap.afterConnect [as oncomplete] (node:ne…]
[JsonrpcGLSPClient] Connection to server is erroring. Shutting down server. (1) [Error: connect ECONNREFUSED 127.0.0.1:5007
	at TCPConnectWrap.afterConnect [as oncomplete] (node:ne…]
Initializing tunnel extension
rejected promise not handled within 1 second: Error: Connection got disposed.
stack trace: Error: Connection got disposed.
	at Object.dispose (/Users/humberto/Downloads/glsp-examples-master 2/project-templates/node-json-vscode/glsp-client/node_modules/vscode-jsonrpc/lib/main.js:876:25)
	at /Users/humberto/Downloads/glsp-examples-master 2/project-templates/node-json-vscode/glsp-client/node_modules/@eclipse-glsp/protocol/lib/jsonrpc/base-jsonrpc-glsp-client.js:73:24
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by Rtulpe** [comment]
Same Issue here, here's the background:
- Windows 10 running on Virtual Box
- VSCode: v1.81.1
- Node.js: v16.17.1, installed using NVM for Windows v1.1.11
- GLSP Set-up: Current node-json-vscode project-template

I could run it (Launch TaskList Diagram Extension) once, then would get the exact same error. I would need to reboot VB instance and run again to see the results. Would also sometimes work again, if I wait for a while.
